product-service/src/utils/product-validate.ts

```diff
@@ -0,0 +1,33 @@
+import { MESSAGES, STATUS_CODES, COMMON_PATH } from '../constants/constants';
+import { IProduct, SKATE_TYPES } from '../models/product.model';
+import { MyError } from './error';
+
+export const productValidate = (product): IProduct => {
```

Comment:
I think that this function is not actually "validates" something. It casts to the correct object.
It's better to describe a schema of desired object. Schema may export 2 methods - validate (which will return errors if validation is not passed) and cast.

Reply:
I will consider this